### PR TITLE
docs: ADR for introducing ICA Controller module on the Hub

### LINKED ISSUE #2869: Enable ICA controller
## Summary

With the [Gaia upgrade to SDK 0.47 and IBC 7](https://github.com/cosmos/gaia/issues/2790), we can enable the ICA controller  on the Cosmos Hub. 

## Problem Definition

The ICA controller would enable the Hub to control accounts on remote chains. An example of an application is the provider-based governance that would allow the governance of a consumer chain to be control by ATOM stakers. See the [IBC docs](https://ibc.cosmos.network/main/apps/interchain-accounts/overview.html) for more details. 

## Proposal

A prerequisite is upgrading the Hub to IBC 7, which will happen with the v15 upgrade. 

## Task list

```[tasklist]
### Must have
- [ ] align with stakeholders on potential use cases (e.g., Neutron)
- [ ] https://github.com/cosmos/gaia/issues/2539
- [ ] https://github.com/cosmos/gaia/issues/2979
```

```[tasklist]
### Nice to have
```
____

#### For Admin Use

- [x] Not duplicate issue
- [x] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

### PR DESCRIPTION
**Pull request by dusan-maksimovic**
## Description

This PR describes how ICA Controller will be introduced on the Hub.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct `docs:` prefix in the PR title
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for thoroughness
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)

### CODE REVIEW

File: docs/docs/architecture/adr/adr-003-ica-controller.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## Consequences
+
+### Positive
+
+Users of the Hub will have a possibility to create and utilize Interchain Accounts on other IBC connected chains.
+
+### Negative
+
+### Neutral
```
Comment (MSalopek) [inline docs/docs/architecture/adr/adr-003-ica-controller.md:54]: Maybe put a few sentences about not having to create custom auth modules to implement ICA (as was previously the case).

File: docs/docs/architecture/adr/adr-003-ica-controller.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+## Abstract
+
+The Interchain Accounts Controller IBC module allows users of one chain to create and control accounts on other chains. The Hub currently doesn't have ICA Controller module enabled, so it is not possible to create accounts on other chains from the Hub chain.
+
+## Context
+
+Enabling the ICA Controller module on the Hub would support various use cases. One such case could be the provider-based governance that would allow the ATOM stakers to participate in a governance on consumer chains.
+
+## Decision
+
```
Comment (MSalopek) [inline docs/docs/architecture/adr/adr-003-ica-controller.md:20]: Maybe add a few details (like code snippets) on the wiring, I think it is enough to put the entire ICA integration from app.go as a diff

File: docs/docs/architecture/adr/adr-003-ica-controller.md (hunk 1 above)
Comment (dusan-maksimovic) [reply docs/docs/architecture/adr/adr-003-ica-controller.md:54]: Added in https://github.com/cosmos/gaia/pull/2994/commits/c5f88348a7b4266afec580c0c871a4b02e8de9df

File: docs/docs/architecture/adr/adr-003-ica-controller.md (hunk 2 above)
Comment (dusan-maksimovic) [reply docs/docs/architecture/adr/adr-003-ica-controller.md:20]: Added in https://github.com/cosmos/gaia/pull/2994/commits/c5f88348a7b4266afec580c0c871a4b02e8de9df

**Review by MSalopek — APPROVED** [review]
LGTM!

File: docs/docs/architecture/adr/adr-003-ica-controller.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# ADR 003: Interchain Accounts Controller Module
+
+## Changelog
+
+- 2024-03-08: Initial Draft
+
+## Status
+
+DRAFT
```
Comment (mpoke) [inline docs/docs/architecture/adr/adr-003-ica-controller.md]:
```suggestion
Proposed
```

**Review by mpoke — APPROVED** [review]

**Comment by mpoke** [comment]
Add an entry to docs/docs/architecture/adr/README.md

**Pull request merged**